py_pdf_parser/visualise/main.py

```diff
@@ -165,7 +165,7 @@ def __plot_current_page(self):
             self.__ax.set_ylim([0, page.height])
 
         page = self.document.get_page(self.current_page)
-        for element in page.elements:
+        for element in page.elements and self.elements:
```

Comment:
```suggestion
        for element in page.elements & self.elements:
```

I think this is not quite correct - `page.elements and self.elements` will return `self.elements` as it's doing a boolean `and` operation (so in the same way that `"a" and "b"` will return `"b"`).

You want to do a set-style intersection of `ElementList`s here, so you want `&` not `and`.